content/en/stateless-apps.md

```diff
@@ -7,30 +7,30 @@ tags: ["fundamental", "application", ""]
-provides functionality to use print services, CDN (Content Delivery Network) or the Web Servers 
-in order to process every short-term request. 
-For example, someone is searching a question in the search engine and pressed the Enter button. 
-In case if the searching operation gets interrupted or closed due to some reason, 
-you have to start a new one as there is no saved data for your previous request.
+Stateless applications process requests as if they were their first. 
+The app doesn't "remember" previous interactions or user session data. 
+Data from previous interactions is referred to as state, and since that data isn't stored anywhere, these apps are state*less*. 
+In short, stateless apps lack permanent storage. 
+Here's an example. 
+When you use a search engine, and that search is interrupted (e.g., the window is closed), those search results are lost. 
+You'll need to start all over.
+
+On the other hand, applications that process requests while considering previous interactions are called [stateful applications](/stateful-apps/). 
 
 ## Problem it addresses
 
-Stateless applications tackle the problem of resiliency, 
-because different pods across a [cluster](/cluster/) can work independently, 
-with multiple requests coming to them at the same time. 
-If there’s a problem, you can easily restart the application, 
-and it will return to its initial state with little or no downtime. 
-As such, the benefits of stateless applications include resiliency, elasticity, and high availability. 
-However, most applications we use today are at least partly [stateful](/stateful-apps/), 
-as they store things like preferences and settings to improve the user experience.
+Stateful applications (applications that *do* save session data) are composed of multiple pods distributed across a [cluster](/cluster/). 
+To function correctly, they must be aware of each other. 
+If one pod processes a request, the other pods must have access to that data to take it into account when they process the next request. 
+That brings some challenges. 
+Pods must constantly communicate to ensure they are all in sync for a seamless user experience. 
+However, networks are inherently unreliable. Also, session data is lost if there’s a problem and the application restarts. 
+In short, stateful apps are less resilient. 
 
 ## How it helps
 
-Boiling everything down, in a Stateless Application the only thing your cluster is responsible for is 
-the code, and other static content, being hosted on it. 
-That’s it, no changing databases, no writes and no left over files when the pod is deleted. 
-Stateless [containers](/container/) are easier to deploy, 
-and you don’t need to worry about saving container data on persistent storage volumes. 
-You also don't have to worry about backing up the data.
+Put simply, in a stateless application, the cluster is only responsible for the code and some static content hosted on it. 
```

Comment:
I don't like how we are putting the focus now on the cluster instead of on the stateful app. Also, stateful and statelessness have nothing to do with a cluster or even the way how the application is deployed, per sé. It just says if an application stores data or not, basically. 

On the other hand, it is kind of hard to say how a stateless app helps. While reviewing and editing, I realized that this is maybe more a concept than a technology and the sections: about the problem it solves and how it helps might be not applicable. 

The stateless application helps by just storing no state, no were. But it is not something that can be simply done for every application. Some need to store data and some not. 

I will think more about it in the next few days, but would like to hear other opinions. @idvoretskyi @CathPag @seokho-son @DaveVentura

Reply:
You are right. This is no different from [tightly coupled](https://glossary.cncf.io/tightly-coupled-architectures/) or [loosely coupled architectures](https://glossary.cncf.io/loosely-coupled-architecture/). It's an architectural decision. How do I build my app?

Reply:
I agree with Noah. It seems a nice technical consideration. 😊

Reply:
In general, I agree with @iamNoah1 

> I don't like how we are putting the focus now on the cluster instead of on the stateful app. Also, stateful and statelessness have nothing to do with a cluster or even the way how the application is deployed, per sé. It just says if an application stores data or not, basically.

I agree. We don't have to introduce about cluster here.

> On the other hand, it is kind of hard to say how a stateless app helps. While reviewing and editing, I realized that this is maybe more a concept than a technology and the sections: about the problem it solves and how it helps might be not applicable.
> 
> The stateless application helps by just storing no state, no were. But it is not something that can be simply done for every application. Some need to store data and some not.

I agree with that utilizing stateless applications or stateful applications is an architectural decision. Both can be helpful according to given requirements or conditions.

I think we can state conditions (best practices?) where stateless application is helpful. (for instance, stateless application is helpful if the application need to handle many short-term requests since it removes overhead in using sessions.)